### PR TITLE
Add problem uploading and approving

### PR DESCRIPTION
**Pull request by Roxanica123**

### CODE REVIEW

File: H3/WebApp/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,9112 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"@angular-devkit/architect@0.901.15":
```
Comment (OloieriAlexandru) [inline H3/WebApp/yarn.lock]: Please remove this. You can add the following rule to a .gitignore file: "**/yarn.lock"

File: H3/WebApp/src/app/app.module.ts [hunk 2]
```diff
@@ -16,7 +15,10 @@ import { MatButtonModule } from '@angular/material/button';
 import { MatIconModule } from '@angular/material/icon';
 import { MatCheckboxModule } from '@angular/material/checkbox';
 import { MatTableModule } from '@angular/material/table';
-
+import  {MatFormFieldModule} from '@angular/material/form-field';
+import {MatInputModule} from '@angular/material/input';
```
Comment (OloieriAlexandru) [inline H3/WebApp/src/app/app.module.ts]: Minor. Indentation is off (glumesc :) )

**Comment by OloieriAlexandru** [comment]
Looks good!

File: H3/WebApp/src/app/models/ProblemApproval.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export class ProblemApproval {
+  readonly approved:boolean;
+  constructor(approved:boolean){
```
Comment (Roxanica123) [inline H3/WebApp/src/app/models/ProblemApproval.ts:3]: space here

File: H3/WebApp/src/app/services/jwt-http-interceptor.service.ts [hunk 4]
```diff
@@ -18,25 +19,29 @@ export class JwtHttpInterceptorService implements HttpInterceptor {
...
-          Accept: 'application/json',
-          'Content-Type': 'application/json',
-        },
-      });
-    }
+    if (token) headers = this.bearerToken(headers, token);
 
-    return next.handle(clone);
+    if (request.url.indexOf('upload') == -1)
+      headers = this.contentType(headers);
```
Comment (OloieriAlexandru) [inline H3/WebApp/src/app/services/jwt-http-interceptor.service.ts]: You should always add curly braces ( `{` + `}` ) for blocks of code (you don't have to change it here, please don't do another commit just for this, but it's a best practice).

File: H3/WebApp/src/app/models/ProblemApproval.ts (hunk 3 above)
Comment (Roxanica123) [reply H3/WebApp/src/app/models/ProblemApproval.ts:3]: I will press CTRL+SHIFT+F more often.

File: H3/WebApp/src/app/services/jwt-http-interceptor.service.ts (hunk 4 above)
Comment (Roxanica123) [reply H3/WebApp/src/app/services/jwt-http-interceptor.service.ts]: Will be solved in the next pull request!
